docs/index.rst

```diff
@@ -421,6 +421,22 @@ Information will be lost but it allows to control any potential performance impa
     tracer.sampler = RateSampler(sample_rate)
 
 
+Resolving deprecation warnings
+------------------------------
+Before upgrading, it’s a good idea to resolve any deprecation warnings raised by your project.
+These warnings must be fixed before upgrading, otherwise ``ddtrace`` library will not work
+as expected. Our deprecation messages include the version where the behavior is altered or
+removed.
+
+In Python, deprecation warnings are silenced by default, and to turn them on you may add the
+following flag or environment variable::
+
+    $ python -Wall app.py
+
+    # or
+
+    $ PYTHONWARNINGS=all python app.py
+
```

Comment:
Yes-ish?

[`logging.captureWarnings`](https://docs.python.org/3/library/logging.html#integration-with-the-warnings-module) and [`warnings.simplefilter`](https://docs.python.org/3/library/warnings.html#updating-code-for-new-versions-of-python) offer some options, but the developer will need to be careful about where those calls are made.

Specifically, they will need to make those calls before any `ddtrace` imports happen (since most of the warnings of interest happen at import time). Additionally, they may have some other import sequencing considerations depending on what their logging infra/setup looks like.

Reply:
What could be the plan here? I may add something programmatic here, but as said by @SeanOC it may not work 100% of the time. I totally agree that devs may not have access to the runtime, but these warnings can still be raised in a local development environment (i.e. running application tests).

If we don't want to provide a programmatic way, I can reword a bit better this part of the documentation explicitly saying what I wrote in the previous paragraph.